lib/dangermattic/plugins/common/common_release_checks.rb

```diff
@@ -0,0 +1,134 @@
+    #
+    # @param release_notes_file [String] The name of the release notes file that should be checked for modifications.
+    #   Example: 'metadata/release_notes.txt'
+    #
+    # @param store_strings_file [String] The name of the store strings file that should be checked for modifications.
+    #   Example: 'metadata/PlayStoreStrings.po'
+    #
+    # @param fail_on_error [Boolean] If true, a failure message will be emitted instead of a warning.
+    #
+    # @example Check if the release notes file 'release_notes.txt' is modified, and the 'PlayStoreStrings.po' file is not modified:
+    #   check_release_notes_and_store_strings(release_notes_file: 'release_notes.txt', store_strings_file: 'PlayStoreStrings.po', fail_on_error: false)
+    #
+    # @return [void]
+    def check_release_notes_and_store_strings(release_notes_file:, store_strings_file:, fail_on_error: false)
+      has_modified_release_notes = danger.git.modified_files.any? { |f| f.end_with?(release_notes_file) }
+      has_modified_app_store_strings = danger.git.modified_files.any? { |f| f.end_with?(store_strings_file) }
+
+      return unless has_modified_release_notes && !has_modified_app_store_strings
+
+      message = "The #{File.basename(store_strings_file)} file must be updated any time changes are made to the release notes."
+
+      if fail_on_error
+        failure(message)
+      else
+        warn(message)
+      end
+    end
+
+    # Check if there are changes to the internal release notes file in the release branch and emit a warning if that's the case.
+    #
+    # @param release_notes_file [String] (optional) The path to the internal release notes file.
+    #        Defaults to the `DEFAULT_INTERNAL_RELEASE_NOTES` constant if not provided.
+    #
+    # @example Checking for changes in the default internal release notes file:
+    #   check_internal_release_notes_changed
+    #
+    # @example Checking for changes in a custom internal release notes file at a specific path:
+    #   check_internal_release_notes_changed(release_notes_file: '/path/to/internal_release_notes.txt')
+    #
+    # @return [void]
+    def check_internal_release_notes_changed(release_notes_file: DEFAULT_INTERNAL_RELEASE_NOTES)
+      warning = <<~WARNING
+        This PR contains changes to `#{release_notes_file}`.
+        Note that these changes won't affect the final version of the release notes as this version is in code freeze.
+        Please, get in touch with a release manager if you want to update the final release notes.
+      WARNING
+
+      check_file_changed(
+        file_comparison: ->(path) { path == release_notes_file },
+        message: warning,
+        on_release: true,
+        fail_on_error: false
+      )
+    end
+
+    private
+
+    def all_modified_files
+      danger.git.added_files + danger.git.modified_files + danger.git.deleted_files
+    end
```

Comment:
This seems common enough to be added too a helper method that could be shared across multiple plugins of this repo?

Also, back when I wrote a couple of Danger rules, I remember that it reports renamed files in a tricky way, i.e. something like when a file is renamed, `danger.git.renamed_files` is set with the list of `{ old_name => new_name }`… but the name used in `danger.git.modified_files` in that case might not be the one you expect (i.e. the old name and not the new name).
Which could lead to red herrings and false positives in our logic (e.g. when we test if file `b.txt` was changed, but what happened in reality is that we renamed `a.txt` to `b.txt`… but `danger.git.modified_files` will list `a.txt`, not `b.txt` so our rule will not detect that `b.txt` was modified…

This is something to address is a separate PR, e.g. when you'll extract that method in some common helper; at that point you might experiment with some real data (e.g. create a real PR that renames some changes and have a `Dangerfile` print the content of `danger.git.added_files.inspect`, `danger.git.modified_files.inspect`, `danger.git.deleted_files.inspect` and `danger.git.renamed_files.inspect`, to see what's returned in reality, so that you can write specs and fixtures for your helper method that reflect that reality, and then make sure it takes those cases into account…?

Reply:
I've turned the class `GitUtils` into a [common plugin and added this util method there](https://github.com/Automattic/dangermattic/blob/17ab16e6c2867b229e080debdd283117a7651229/lib/dangermattic/plugins/common/git_utils.rb).
Later I've also [renamed](https://github.com/Automattic/dangermattic/blob/24943497dfd32fb1ee7d573b4555c67da268bc7a/lib/dangermattic/plugins/common/git_utils.rb) it to `all_changed_files` because the name per se was confusing (similar to the `modified_files` used by the Git API as well).

The case you mentioned about renames, tbh I'm not sure yet, but that's a good point for me to test as well 👍